### PR TITLE
Rewrote VM

### PR DESCRIPTION
**Pull request by stormprograms**
* removed `do`
* fixed a ton of issues as highlighted in the change log
* new version number: v1.0.0 Beta 1

### CODE REVIEW

File: src/values/compare.cc [hunk 1]
```diff
@@ -1,133 +1,79 @@
...
-    0: var1
-    1: comparison operator
-    2: var2
-    3: [
-    4: n
-    5: ]
+    The right and left hand values can also contain subscripts, in which case they would look more like this:
+    UNKNOWN | TOKEN | VALUE/UNKNOWN | TOKEN
+    var     |  [    |     INTEGER   |   ]
     */
```
Comment (Metr0Gn0me) [inline src/values/compare.cc]:
```suggestion
```

File: src/parser/parse.cc [hunk 2]
```diff
@@ -3,7 +3,7 @@
 the interpreter parses the file and calls functions in other files
 */
```
Comment (Metr0Gn0me) [inline src/parser/parse.cc]:
```suggestion
```

File: src/interpreter/eval.cc [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    else if ((kwd == "random") || (kwd == "randomrange") || (kwd == "length")) return 1;
+    return 0;
+}
+
+bool evalLiteral(string ctn) {
+    if ((ctn.front() == '\"') || (isint(ctn)) || (ctn == "true") || (ctn == "false")) return 1;
+
+    return 0;
+}
+
```
Comment (Metr0Gn0me) [inline src/interpreter/eval.cc]:
```suggestion
```

File: src/include/core.h [hunk 4]
```diff
@@ -5,72 +5,85 @@
...
+	void declareFunc(int *y, std::vector<stsfunc> *functions);
+
+	void wait(std::vector<stsvars> *vars, std::vector<stsfunc> functions, int y) {
+		#if PLATFORM == 1
+			Sleep(std::stoi(getval(vars, functions, new int(y+1)).val) * 1000);
+		#else
+			sleep(std::stoi(getval(vars, functions, new int(y+1)).val));
+		#endif
+	}
 };
```
Comment (Metr0Gn0me) [inline src/include/core.h]:
```suggestion
```

File: tests/outputs/concattest.sts.txt [hunk 5]
```diff
@@ -1 +1 @@
-Hi bob
+200 is my favorite number
```
[the remark this answers is not in the document]
Comment (stormprograms) [reply tests/outputs/concattest.sts.txt]: ...

**Comment by stormprograms** [comment]
Thank you for fixing my white space problem

**Pull request merged**